src/components/BubbleDisplay/BubbleDisplay.js

```diff
@@ -23,7 +23,55 @@ class BubbleDisplay extends Component {
   };
 
   render() {
-    return <div />;
+    const { width, height, zoom, x, points, children } = this.props;
+    const realWidth = width * zoom;
+    const maxWidth = Math.max.apply(null, points.map(point => point.to));
+    const maxDepth = 3; //TODO: compute max depth
+    const projectionFactor = realWidth / maxWidth;
+    const viewBox = [x * projectionFactor, 0, width, height].join(' ');
+    const bubbles = points.map(point => ({
+      x: point.from * projectionFactor,
+      width: (point.to - point.from) * projectionFactor,
+      colour: point.colour,
+      height: Math.pow(2 / 3, point.depth - 1) * height,
+      label: point.label,
+      point: point,
+    }));
```

Comment:
We could move the generation of `realWidth`, `maxDepth`, `maxWidth`, `projectionFactor`, `viewBox` and `bubbles` into `componentWillMount` and `componentWillReceiveProps` later on, and only re-render when the data changes. Might be needed later on if we're doing a bunch of updates to this (dragging, panning, animating etc.) No need to do it just now though, something for later